tortoise/util.go

```diff
@@ -81,26 +73,24 @@ func (a vec) String() string {
 	return "abstain"
 }
 
-func calculateOpinionWithThreshold(logger log.Log, v vec, layerSize int, theta uint8, delta float64) vec {
-	threshold := float64(theta) / 100 * delta * float64(layerSize)
-	netVote := float64(v.netVote())
+func calculateOpinionWithThreshold(logger log.Log, v vec, layerSize int, theta uint8, delta uint32) vec {
+	threshold := big.NewFloat(float64(theta))
+	threshold = threshold.Quo(threshold, big.NewFloat(100))
+	threshold = threshold.Mul(threshold, big.NewFloat(float64(delta)*float64(layerSize)))
```

Comment:
Wouldn't this be more precise because of avoiding `float64` multiplication?

```suggestion
	threshold = threshold.Mul(threshold, big.NewFloat(float64(delta)))
	threshold = threshold.Mul(threshold, big.NewFloat(float64(layerSize)))
```

Reply:
delta and layerSize are integers, in what case there will be issues with precision?

Reply:
Sorry, I looked at the previous signature of `calculateOpinionWithThreshold`. Never mind

Reply:
Although, I think the precision might be lost just a little bit in this case because of converting to `float64`. `float64(delta)` would become some floating-point value around `delta`, then it would be multiplied by something around `layerSize` and the result would be something around `delta*layerSize` (same for `theta` and `100`). `new(big.Float).SetUint64(uint64(delta))` should avoid losing precision.

Reply:
I wrote a test for this, and it shows that I'm wrong
```
func Test_Precision(t *testing.T) {
	var (
		theta     uint8  = 2
		layerSize int    = 5
		delta     uint32 = 10
	)

	t1 := big.NewFloat(float64(theta))
	t1 = t1.Quo(t1, big.NewFloat(100))
	t1 = t1.Mul(t1, big.NewFloat(float64(delta)*float64(layerSize)))

	t2 := new(big.Float).SetUint64(uint64(theta))
	t2 = t2.Quo(t2, new(big.Float).SetUint64(100))
	t2 = t2.Mul(t2, new(big.Float).SetUint64(uint64(delta)))
	t2 = t2.Mul(t2, new(big.Float).SetUint64(uint64(layerSize)))

	t.Log(t1)
	t.Log(t2)
	require.Equal(t, t1.Cmp(t2), 0)
}
```
Output:
```
=== RUN   Test_Precision
    util_test.go:26: 1
    util_test.go:27: 0.99999999999999999995
    util_test.go:28: 
        	Error Trace:	util_test.go:28
        	Error:      	Not equal: 
        	            	expected: 1
        	            	actual  : 0
        	Test:       	Test_Precision
```